Classes/Utility/UserUtility.php

```diff
@@ -246,18 +246,22 @@ public static function getDirtyPropertiesFromUser(User $changedObject)
                 && !in_array($propertyName, $ignoreProperties)
             ) {
                 $newPropertyValue = $changedObject->{'get' . ucfirst($propertyName)}();
-                if (!is_object($oldPropertyValue) || !is_object($newPropertyValue)) {
+                if (!is_object($oldPropertyValue) && !is_object($newPropertyValue)) {
                     if ($oldPropertyValue !== $newPropertyValue) {
                         $dirtyProperties[$propertyName]['old'] = $oldPropertyValue;
                         $dirtyProperties[$propertyName]['new'] = $newPropertyValue;
                     }
                 } else {
-                    if (get_class($oldPropertyValue) === 'DateTime') {
+                    if (($oldPropertyValue != null && get_class($oldPropertyValue) === 'DateTime') || ($newPropertyValue != null && get_class($newPropertyValue) === 'DateTime')) {
```

Comment:
Please use type safe comparisons (`!==`)

Reply:
Also, use `$oldPropertyValue instanceof \DateTime::class` instead of `get_class`.

Reply:
Added your suggestions, but i left out the `::class` since `\DateTime` is already the fully qualified class name.